### PR TITLE
scripts: Create script to copy contents from previous minor version to current

### PR DESCRIPTION
**Pull request by piyush-jena**
**Issue number:**

Closes #

**Description of changes:**
- This new script safely copies contents from previous minor version to next. 

**Terms of contribution:**

By submitting this pull request, I confirm that my contribution is made under
the terms of the licenses outlined in the LICENSE-SUMMARY file.

### CODE REVIEW

File: scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# Copy previous minor version content to the latest minor version
+cp -R $BOTTLEROCKET_WEBSITE_REPO_DIR/content/en/os/$PREV_MAJOR_MINOR_X/ \
+$BOTTLEROCKET_WEBSITE_REPO_DIR/content/en/os/$MAJOR_MINOR_X/
+cp -R $BOTTLEROCKET_WEBSITE_REPO_DIR/data/settings/$PREV_MAJOR_MINOR_X/ \
+$BOTTLEROCKET_WEBSITE_REPO_DIR/data/settings/$MAJOR_MINOR_X/
+
+# Remove invalid files and folders
+rm -rf $BOTTLEROCKET_WEBSITE_REPO_DIR/content/en/os/$MAJOR_MINOR_X/version-information/packages/$MAJOR.*
+rm -rf $BOTTLEROCKET_WEBSITE_REPO_DIR/content/en/os/$MAJOR_MINOR_X/version-information/gpu-drivers/$MAJOR.*
```
Comment (ginglis13) [inline scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh:48]:
```suggestion
cp -R "${BOTTLEROCKET_WEBSITE_REPO_DIR}/content/en/os/${PREV_MAJOR_MINOR_X}/" \
"${BOTTLEROCKET_WEBSITE_REPO_DIR}/content/en/os/${MAJOR_MINOR_X}/"
cp -R "${BOTTLEROCKET_WEBSITE_REPO_DIR}/data/settings/"${PREV_MAJOR_MINOR_X}/" \
"${BOTTLEROCKET_WEBSITE_REPO_DIR}/data/settings/$:MAJOR_MINOR_X}/"

# Remove invalid files and folders
rm -rf "${BOTTLEROCKET_WEBSITE_REPO_DIR}/content/en/os/${MAJOR_MINOR_X}/version-information/packages/${MAJOR}.*"
rm -rf "${BOTTLEROCKET_WEBSITE_REPO_DIR}/content/en/os/${MAJOR_MINOR_X}/version-information/gpu-drivers/${MAJOR}.*"
```

File: scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
```
Comment (ginglis13) [inline scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh:1]: let's use `${VAR}` over `$VAR` https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion

File: scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+PREV_MAJOR_MINOR_X="$MAJOR.$PREV_MINOR.x"
+
+# Copy previous minor version content to the latest minor version
+cp -R $BOTTLEROCKET_WEBSITE_REPO_DIR/content/en/os/$PREV_MAJOR_MINOR_X/ \
+$BOTTLEROCKET_WEBSITE_REPO_DIR/content/en/os/$MAJOR_MINOR_X/
+cp -R $BOTTLEROCKET_WEBSITE_REPO_DIR/data/settings/$PREV_MAJOR_MINOR_X/ \
+$BOTTLEROCKET_WEBSITE_REPO_DIR/data/settings/$MAJOR_MINOR_X/
+
+# Remove invalid files and folders
+rm -rf $BOTTLEROCKET_WEBSITE_REPO_DIR/content/en/os/$MAJOR_MINOR_X/version-information/packages/$MAJOR.*
```
Comment (larvacea) [inline scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh:47]: variable expansions should ideally be quoted.

**Review by larvacea — COMMENTED** [review]
I recommend `shellcheck` for all bash scripts.

It's available as an extension in Visual Studio Code (easy mode) or as a standalone tool (via https://github.com/koalaman/shellcheck)

**Comment by piyush-jena** [comment]
^ Resolved comments

**Review by larvacea — APPROVED** [review]

File: scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+# Path to website repository
+BOTTLEROCKET_WEBSITE_REPO_DIR="$(dirname "$(dirname "$(dirname "$(realpath "$0")")")")"
+
+# Check if the correct number of arguments is provided
+if [ "$#" -ne 1 ]; then
+  echo "Usage: $0 <version>"
+  exit 1
+fi
+
+VERSION=$1
```
Comment (ginglis13) [inline scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh:12]:
```suggestion
VERSION="${1}"
```

File: scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+# Check if the version format is valid (X.Y.Z)
+if ! [[ $VERSION =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
```
Comment (ginglis13) [inline scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh:15]:
```suggestion
if ! [[ "${VERSION}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
```

File: scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  echo "Invalid version format. Expected: X.Y.Z"
+  exit 1
+fi
+
+MAJOR=$(echo "${VERSION}" | cut -d'.' -f1)
+MINOR=$(echo "${VERSION}" | cut -d'.' -f2)
+PATCH=$(echo "${VERSION}" | cut -d'.' -f3)
+
+# Check if MINOR is 0, which is an invalid case
+if [ "$MINOR" -eq 0 ]; then
```
Comment (ginglis13) [inline scripts/website-boiler-plate-generator/copy_files_from_prev_version.sh:25]:
```suggestion
if [ "${MINOR}" -eq 0 ]; then
```

**Review by ginglis13 — APPROVED** [review]

**Comment by piyush-jena** [comment]
^ Fixed comments

**Pull request merged**